src/test/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProviderTest.java

```diff
@@ -58,139 +42,235 @@ public static Collection<Object[]> input() {
 
   public StreamingClientProviderTest(boolean enableClientOptimization) {
     this.enableClientOptimization = enableClientOptimization;
-  }
-
-  @Before
-  public void setup() {
-    // setup fresh configs
-    this.clientConfig1 = TestUtils.getConfForStreaming();
-    this.clientConfig1.put(
+    this.clientConfig.put(
         SnowflakeSinkConnectorConfig.ENABLE_STREAMING_CLIENT_OPTIMIZATION_CONFIG,
-        this.enableClientOptimization + "");
-    this.clientConfig2 = new HashMap<>(this.clientConfig1);
-
-    this.clientConfig1.put(Utils.NAME, "client1");
-    this.clientConfig2.put(Utils.NAME, "client2");
-
-    this.streamingClientHandler = Mockito.spy(StreamingClientHandler.class);
-    this.streamingClientProvider =
-        StreamingClientProvider.getStreamingClientProviderForTests(
-            null, this.streamingClientHandler);
-  }
-
-  @After
-  public void tearDown() {
-    this.streamingClientHandler.closeClient(this.client1);
```

Comment:
Since this is UT it should not actually create the client, so I refactored this test to use mocks and pulled most of the complexity into the new IT. caught by @sfc-gh-japatel